build.gradle

```diff
@@ -47,65 +45,83 @@ dependencies {
 }
 
 
-task sourceJar(type: Jar) { from sourceSets.main.allJava }
-
-publishing {
-	publications {
-		mavenJava(MavenPublication) {
-			from components.java
-			
-			artifact sourceJar { classifier "sources" }
-		}
-	}
-	repositories { 
-		maven { 
-			url "../mvn-repo" 
-		} 
-	}
+task javadocJar(type: Jar) {
+    classifier = 'javadoc'
+    from javadoc
 }
 
-// Prevent Eclipse "Gradle -> Refresh All" to execute cleanEclipseJdt,
-// because it would delete our version of the org.eclipse.jdt.ui.prefs file.
-cleanEclipse.dependsOn = cleanEclipse.taskDependencies.values - cleanEclipseJdt
-
-
-configurations { xjc }
+task sourcesJar(type: Jar) {
+    classifier = 'sources'
+    from sourceSets.main.allSource
+}
 
-dependencies {
-	xjc group: 'com.sun.xml.bind', name: 'jaxb-xjc', version: '2.2.4-1'
+artifacts {
+    archives javadocJar, sourcesJar
 }
 
-project.ext.schemaTargetDir = new File("$projectDir/src/main/resources/scenarioo-schemas");
-task createSchemaDirectory << {
-	println 'creating schema directory: ' + schemaTargetDir
-    schemaTargetDir.mkdirs()
-    if (schemaTargetDir.isDirectory()) {
-        println 'schemas directory created properly.'
-    }
-    else {
-    	println 'ERROR: creating schema directory failed.'
-    	throw new MissingResourceException("Schema directory was not present and could not be created")
-    }
+signing {
+    sign configurations.archives
 }
 
-task removeSchemaDirectory(type: Delete) {
-  delete schemaTargetDir
+tasks.withType(Javadoc) {
+    options.addStringOption('Xdoclint:none', '-quiet')
 }
 
-task schemagen(dependsOn: 'createSchemaDirectory') << {
-    ant.taskdef(name: 'schemagen', classname: 'com.sun.tools.jxc.SchemaGenTask', classpath: configurations.xjc.asPath)
+uploadArchives {
+    repositories {
+        mavenDeployer {
+            beforeDeployment { MavenDeployment deployment -> signing.signPom(deployment) }
+
+            repository(url: "https://oss.sonatype.org/service/local/staging/deploy/maven2/") {
+                authentication(userName: ossrhUsername, password: ossrhPassword)
+            }
+
+            snapshotRepository(url: "https://oss.sonatype.org/content/repositories/snapshots/") {
+                authentication(userName: ossrhUsername, password: ossrhPassword)
+            }
+
+            pom.project {
+                name 'Scenarioo java'
+                packaging 'jar'
+                description 'Scenarioo Writer library for Java to generate Scenarioo Documentation http://www.scenarioo.org'
+                url 'http://www.scenarioo.org'
+
+                scm {
+                    connection 'https://github.com/scenarioo/scenarioo-java.git'
+                    developerConnection 'https://github.com/scenarioo/scenarioo-java.git'
+                    url 'https://github.com/scenarioo/scenarioo-java.git'
+                }
+
+                licenses {
+                    license {
+                        name 'GNU GENERAL PUBLIC LICENSE'
```

Comment:
name of the licence is wrong, since we have the linking exception, see https://en.wikipedia.org/wiki/GPL_linking_exception  - so it should probably be "GNU GENERAL PUBLIC LICENCE with linking exception"